### PR TITLE
blas,lapack: amend libblas, libcblas, liblapack, liblapacke for darwin with mkl

### PR DESCRIPTION
**Pull request by sepiabrown**
###### Description of changes

When `blasProvider` and `lapackProvider` are assigned to be `mkl`,
`${lib.get Lib blasProvider'}/lib/libblas.dylib`, `${lib.get Lib blasProvider'}/lib/libcblas.dylib`, `${lib.get Lib lapackProvider'}/lib/liblapack.dylib`, `${lib.get Lib lapackProvider'}/lib/liblapacke.dylib` don't exist.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [-] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [-] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

I've tested with iMac 2014 Big Sur. 20000 by 20000 matrix norm was calculated which took about 76 seconds.

### CODE REVIEW

**Comment by sepiabrown** [comment]
> How is this handled on Linux? 
When `blasProvider` and `lapackProvider` are assigned to be `mkl`,
`${lib.get Lib blasProvider'}/lib/libblas.dylib`, `${lib.get Lib blasProvider'}/lib/libcblas.dylib`, `${lib.get Lib lapackProvider'}/lib/liblapack.dylib`, `${lib.get Lib lapackProvider'}/lib/liblapacke.dylib` does exist for Linux.

> Would it instead be smarter to rearrange the `mkl` derivation to put the libraries in the correct place?

Since `libblas` and `liblapack` variables are used throughout the file, I couldn't think of a better solution than the above.

File: pkgs/build-support/alternatives/blas/default.nix [hunk 1]
```diff
@@ -65,7 +65,11 @@ stdenv.mkDerivation {
   installPhase = (''
   mkdir -p $out/lib $dev/include $dev/lib/pkgconfig
 
-  libblas="${lib.getLib blasProvider'}/lib/libblas${canonicalExtension}"
+  if [ "${builtins.toString stdenv.hostPlatform.isDarwin}" == "1" ] && [ "${blasImplementation}" == "mkl" ]; then
```
Comment (ttuegel) [inline pkgs/build-support/alternatives/blas/default.nix:68]: This is the way we usually write these conditionals in Nixpkgs:

```suggestion
  if [ -n "${builtins.toString stdenv.hostPlatform.isDarwin}" ] && [ "${blasImplementation}" == "mkl" ]; then
```

**Review by ttuegel — CHANGES_REQUESTED** [review]
This would cause a mass rebuild. I would suggest moving the conditional from the Bash level to the Nix level. If that isn't possible, then this PR needs to target the `staging` branch.

**Comment by sepiabrown** [comment]
Opss, I will reopen the issue to `staging` branch.

**Comment by mweinelt** [comment]
A good time to read up on how to do a rebase without requesting most every codeowner for review. It's part of the CONTRIBUTING.md.

https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#rebasing-between-branches-ie-from-master-to-staging

**Comment by sepiabrown** [comment]
> A good time to read up on how to do a rebase without requesting most every codeowner for review. It's part of the CONTRIBUTING.md.
> 
> https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#rebasing-between-branches-ie-from-master-to-staging

Thank you for the information!

**Pull request closed without merging**